### PR TITLE
Wait apiserver in controller manager startup process

### PR DESCRIPTION
**Pull request by deyuanix**
I've seen multiple times that controller manager failed to start because of unable to contact apiserver.

### CODE REVIEW

**Comment by ncdc** [comment]
@kubernetes/rh-cluster-infra @liggitt @deads2k @smarterclayton

File: cmd/kube-controller-manager/app/controllermanager.go [hunk 1]
```diff
@@ -309,7 +309,17 @@ func (s *CMServer) Run(_ []string) error {
 
 	resourcequotacontroller.NewResourceQuotaController(kubeClient).Run(s.ResourceQuotaSyncPeriod)
 
-	versionStrings, err := client.ServerAPIVersions(kubeconfig)
+	// If apiserver is not running we should wait for some time and fail only then. This is particularly
+	// important when we start apiserver and controller manager at the same time.
+	var versionStrings []string
+	for i := 0; i < 10; i++ {
```
Comment (ashcrow) [inline cmd/kube-controller-manager/app/controllermanager.go]: Just curious, is there a specific reason for 10 attempts or is it a reasonable guess?

Comment (deads2k) [inline cmd/kube-controller-manager/app/controllermanager.go]: This reads a lot like `wait.PollImmediate`.

Comment (smarterclayton) [inline cmd/kube-controller-manager/app/controllermanager.go]: Yeah, would prefer poll immediate.

Comment (ddysher) [inline cmd/kube-controller-manager/app/controllermanager.go]: Nice pointer.

@ashcrow 10 was copied from iptable/controller.  I think it's just a wild guess?  BTW, I've also changed it to wait.PollImmediate.

**Comment by davidopp** [comment]
Reassigning to @lavalamp

**Comment by lavalamp** [comment]
LGTM

**Comment by lavalamp** [comment]
Am willing to accept a cherrypick if it's easy.

**Pull request merged**